---
title: "Report Reference Samples"
author: "Alejandra Hernandez Segura"
date: "`r format(Sys.Date(), '%d %B %Y')`"
output: 
  html_document:
    toc: true
    toc_float: true
    toc_depth: 2
    self_contained: yes
editor_options: 
  chunk_output_type: inline
params:
  input_dir:
    value: "C:/Users/alehd/Documents/Test_refsamp/Juno_results/Juno_results/200922_NB502001_0153_AHGNLNAFX2_0007/"
  history_data: 
    value: "C:/Users/alehd/Documents/Test_refsamp/out/history_data.csv"
---

<style>
body {
text-align: justify}
</style>


```{r setup, include=FALSE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo = FALSE, 
                      warning = FALSE, message = FALSE,
                      fig.align = "center", fig.retina = 2,
                      fig.height = 4, fig.width = 7)

#Load libraries 
library(dplyr)
library(ggplot2)
library(scales)
library(refsamp)

```

**Run folder:** ```r basename(params$input_dir)```

```{r load_data}

quast_report <- extract_quast(params$input_dir)

checkm_report <- extract_checkm(params$input_dir)

bbtools_report <- extract_bbtools(params$input_dir)

run_date <- get_run_date(params$input_dir)

```

```{r refsamp_run_metrics}

refsamp_run_metrics <- merging_by_sample(c("bbtools_report", "quast_report", "checkm_report"), run_date = run_date)

if ( file.exists(params$history_data) ) {
  
  history_data <- readr::read_csv(params$history_data, col_types = "cnnnnnnnnnnncc") %>%
    bind_rows(refsamp_run_metrics) %>%
    distinct() %>%
    mutate(Run_date = as.Date(.$Run_date, format = "%y-%m-%d")) %>%
    filter(difftime(Run_date, Sys.Date(), units = "days") %>% as.numeric() %>% `>`(-180))
  
} else {
  
  history_data <- refsamp_run_metrics
  
}

```

A set of reference samples are continuously sequenced in our in house sequencing facility and the assembly metrics are monitored to ensure quality of the data produced. All samples were pre-processed and their quality was analyzed using the [Juno pipeline](https://github.com/AleSR13/Juno_pipeline). Some of the quality metrics are analyzed and followed over time.

**List of reference samples:**

```{r refsamp_genus, out.width="30%"}

refsamp:::refsamp_genus %>% knitr::kable()

```

## Quality assessment statistics with QUAST

### Number of contigs

Number of contigs
  : total number of contigs in the assembly [REF](https://academic.oup.com/bioinformatics/article/29/8/1072/228832?login=true).

In the Juno pipeline, scaffolds smaller than 500 nt are filtered out. This plot shows the number of contigs that were kept in the pipeline.

```{r plot_contigs}

plot_time_metrics(history_data, "# contigs") +
  labs(y = "Number of contigs ( >500nt )", x = "") +
   scale_y_continuous( label = comma_format(accuracy = 0.1) )

```

### Total length

Total length
  : Total number of bases in the assembly [REF](https://academic.oup.com/bioinformatics/article/29/8/1072/228832?login=true). 
```{r plot_length}

plot_time_metrics(history_data, "Total length") + 
   labs(y = "Total length", x = "") +
   scale_y_continuous( label = comma_format(accuracy = 1) )
 
```

### N50

N50 
  : The largest contig length, L, such that using contigs of length forumla accounts for at least 50% of the bases of the assembly [REF](https://academic.oup.com/bioinformatics/article/29/8/1072/228832?login=true). 
  
The N50 relates to contiguity of the assembly. The N50 value is a threshold contig size. If contigs with size N50 or larger are kept, half of the reference genome can be covered [REF](http://www.metagenomics.wiki/pdf/definition/assembly/n50).

```{r plot_N50}

plot_time_metrics(history_data, "N50") + 
  labs(y = "N50 (bp)", x = "") +
   scale_y_continuous( label = comma_format(accuracy = 1) )

```

### L50

L50
  : Smallest number of contigs whose length sum makes up half of genome size [REF](https://en.wikipedia.org/wiki/N50,_L50,_and_related_statistics).

```{r plot_L50}

plot_time_metrics(history_data, "L50") + 
  labs(y = "L50", x = "") +
   scale_y_continuous( label = comma_format(accuracy = 1) )

```

### GC content

GC content
  : percentage of nucleotides that are either G or C. 
  
A deviation from the actual (literature reported) GC content in the reference genome can reflect problems in the sequencing of that particular genus/species, contamination or even swapping of samples. 

```{r}

plot_time_metrics(history_data, "GC (%)") + 
  labs(x = NULL, y = "GC content") +
  scale_y_continuous( label = percent_format(accuracy = 0.1, scale = 1) )
  

```


## Quality assessment statistics with Bbtools

### Percent mapped

Percent mapped
  : percentage of reads that could be mapped back to the assembly.

```{r plot_perc_mapped}

plot_time_metrics(history_data, "Percent mapped") + 
  labs(x = NULL, y = "Mapped Reads")+
   scale_y_continuous( label = percent_format(accuracy = 0.1, scale = 1) )

```

### Average depth of coverage 

Coverage, depth of coverage or mapping depth
  : average number of times a base of a genome is sequenced. Calculated as the number of bases of all short reads that match a genome divided by the length of this genome. It is often expressed as 1X, 2X, 3X,... (1, 2, or, 3 times coverage) [REF](http://www.metagenomics.wiki/pdf/definition/coverage-read-depth).

```{r plot_coverage}

plot_time_metrics(history_data, "Average coverage") + 
  labs(x = NULL, y = "Average coverage") +
   scale_y_continuous( label = comma_format(accuracy = 1, suffix = " X") )

```

### Percent of reference bases covered

Percent of reference bases covered 
  : percentage of bases of a reference genome that are covered by at least 1 read. 


```{r plot_perc_covered}

plot_time_metrics(history_data, "Percent of reference bases covered") + 
  labs(x = NULL, y = "Reference Bases Covered")+
  scale_y_continuous( label = percent_format(accuracy = 0.1, scale = 1) )

```

## Quality assessment statistics with CheckM

CheckM makes use of a set of taxon-specific marker genes. It calculates statistics based on that.

### Contamination

Contamination
  : estimated from the number of multicopy marker genes identified in each marker set [REF](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4484387/).

```{r plot_contamination}

plot_time_metrics(history_data, "contamination") + 
  labs(x = NULL, y = "Contamination")+
   scale_y_continuous( label = percent_format(accuracy = 0.01, scale = 1) )

```

### Genome completeness

Genome completeness
  : estimated as the number of marker sets present in a genome taking into account that only a portion of a marker set may be identified [REF](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4484387/).

```{r plot_completeness}

plot_time_metrics(history_data, "completeness") + 
  labs(x = NULL, y = "Completeness") +
   scale_y_continuous( label = percent_format(accuracy = 0.1, scale = 1) )

```


```{r save_history_data}

write.csv(history_data, paste0(params$history_data), row.names = FALSE)

```
